### ISSUE TITLE
[Errno 13] Permission denied: 'dvipng'

### ISSUE THREAD

**Issue by Cynthial0l** [tanghaibao/jcvi#408]
Hi,

I run the command `python -m jcvi.compara.catalog ortholog ser pam --no_strip_names`. It seems to be running normally, but in the end the following error was reported.

> [04:29:58 PM] DEBUG    Dot plot title: Inter-genomic comparison: Cser vs Pcam (18,153 gene pairs)                                       dotplot.py:381
[04:29:59 PM] ERROR    savefig failed. Reset usetex to False.                                                                              base.py:287
                       [Errno 13] Permission denied: 'dvipng'                                                                                         
Traceback (most recent call last):
  File "/home/miniconda3/lib/python3.7/site-packages/jcvi/graphics/base.py", line 283, in savefig
    plt.savefig(figname, dpi=dpi, format=format)
  File "/home/miniconda3/lib/python3.7/site-packages/matplotlib/pyplot.py", line 967, in savefig
    fig.canvas.draw_idle()   # need this if 'transparent=True' to reset colors
  File "/home/miniconda3/lib/python3.7/site-packages/matplotlib/backend_bases.py", line 2055, in draw_idle
    self.draw(*args, **kwargs)
  File "/home/miniconda3/lib/python3.7/site-packages/matplotlib/backends/backend_agg.py", line 406, in draw
    self.figure.draw(self.renderer)
  File "/home/miniconda3/lib/python3.7/site-packages/matplotlib/artist.py", line 74, in draw_wrapper
    result = draw(artist, renderer, *args, **kwargs)
  File "/home/miniconda3/lib/python3.7/site-packages/matplotlib/artist.py", line 51, in draw_wrapper
    return draw(artist, renderer, *args, **kwargs)
  File "/home/miniconda3/lib/python3.7/site-packages/matplotlib/figure.py", line 2791, in draw
    renderer, self, artists, self.suppressComposite)
  File "/home/miniconda3/lib/python3.7/site-packages/matplotlib/image.py", line 132, in _draw_list_compositing_images
    a.draw(renderer)
  File "/home/miniconda3/lib/python3.7/site-packages/matplotlib/artist.py", line 51, in draw_wrapper
    return draw(artist, renderer, *args, **kwargs)
  File "/home/miniconda3/lib/python3.7/site-packages/matplotlib/_api/deprecation.py", line 431, in wrapper
    return func(*inner_args, **inner_kwargs)
  File "/home/miniconda3/lib/python3.7/site-packages/matplotlib/axes/_base.py", line 2921, in draw
    mimage._draw_list_compositing_images(renderer, self, artists)
  File "/home/miniconda3/lib/python3.7/site-packages/matplotlib/image.py", line 132, in _draw_list_compositing_images
    a.draw(renderer)
  File "/home/miniconda3/lib/python3.7/site-packages/matplotlib/artist.py", line 51, in draw_wrapper
    return draw(artist, renderer, *args, **kwargs)
  File "/home/miniconda3/lib/python3.7/site-packages/matplotlib/text.py", line 725, in draw
    mtext=mtext)
  File "/home/miniconda3/lib/python3.7/site-packages/matplotlib/backends/backend_agg.py", line 255, in draw_tex
    Z = texmanager.get_grey(s, size, self.dpi)
  File "/home/miniconda3/lib/python3.7/site-packages/matplotlib/texmanager.py", line 364, in get_grey
    pngfile = self.make_png(tex, fontsize, dpi)
  File "/home/miniconda3/lib/python3.7/site-packages/matplotlib/texmanager.py", line 352, in make_png
    self._run_checked_subprocess(cmd, tex)
  File "/home/miniconda3/lib/python3.7/site-packages/matplotlib/texmanager.py", line 254, in _run_checked_subprocess
    stderr=subprocess.STDOUT)
  File "/home/miniconda3/lib/python3.7/subprocess.py", line 395, in check_output
    **kwargs).stdout
  File "/home/miniconda3/lib/python3.7/subprocess.py", line 472, in run
    with Popen(*popenargs, **kwargs) as process:
  File "/home/miniconda3/lib/python3.7/subprocess.py", line 775, in __init__
    restore_signals, start_new_session)
  File "/home/miniconda3/lib/python3.7/subprocess.py", line 1522, in _execute_child
    raise child_exception_type(errno_num, err_msg, err_filename)
PermissionError: [Errno 13] Permission denied: 'dvipng'

Any help would be appreciated!

Thanks and regards

**Comment by tanghaibao** [comment]
@Cynthial0l 

You are possibly missing LaTex. Take a look at the FAQ section:

https://github.com/tanghaibao/jcvi/wiki

Alternatively, add option `--notex` to the dot plot command, which is the step (last step of the `ortholog` pipeline) you are failing.

```console
python -m jcvi.graphics.dotplot ser.pam.anchors --notex
```

**Comment by Cynthial0l** [comment]
Got it! Thanks a lot!

**Comment by ohdongha** [comment]
Hello @tanghaibao,
Sorry to freeride on someone else's closed issue.  I vaguely remember `tex` was mentioned as a dependency (like `lastal`) when I installed MCscan-jcvi last time, but not now.   I also noticed that even there is this "Permission denied: 'dvipng'" error, the dotplot, etc. were still plotted.  And there is an option not to use `tex` at all. 

Is  `tex` needed only for using, say, Helvetica font (I remember this font was mentioned during the interaction with MCscan without `--notex`)?  What would be the difference between the output with or without `tex`? 

It is great to have an option not to use `tex` because sometimes it is a bit heavy to install, especially on the school HPC server with limited `/home` space.

Thanks!
Dong-Ha
